ui/src/components/questions/post.js

```diff
@@ -0,0 +1,277 @@
+        }
+
+        return t`Delete question`;
+    }
+
+    getDialogBody() {
+        let prefix = <b>{this.props.title}</b>;
+        if (!this.props.questionId) {
+            prefix = t`The answer`;
+        }
+
+        return <>{prefix} {t`will be deleted from Rocket fuel.`}</>;
+    }
+
+    editPost() {
+        if (!this.props.questionId) {
+            this.setState({
+                answer: this.props.body,
+                isEditDialogOpen: true
+            });
+            return;
+        }
+        this.props.history.push(`/create/thread/${this.props.questionId}`);
+    }
+
+    saveAnswer() {
+        if (!this.props.answerId) {
+            return;
+        }
+        this.setState({
+            postingToServer: true
+        });
+        User.updateAnswer(this.props.answerId, {answer: this.state.answer})
+            .then(() => {
+                this.setState({
+                    isEditDialogOpen: false,
+                    currentBody: this.state.answer,
+                    postingToServer: false
+                });
+                this.props.onEdit();
+            }).catch(() => {
+            this.setState({
+                postingToServer: false
+            });
+        });
+    }
+
+    delete() {
+        if (!this.props.questionId) {
+            User.deleteAnswer(this.props.answerId).then(() => {
+                this.setState({
+                    isDeleteDialogOpen: false
+                });
+                this.props.onDelete();
+            });
+            return;
+        }
+        User.deleteQuestion(this.props.questionId).then(() => {
+            this.setState({
+                isDeleteDialogOpen: false
+            });
+            this.props.onDelete();
+            this.props.history.replace('/');
+        });
+    }
+
+    renderVotes() {
+        return (
+            <>
+                <div className="vote">
+                    <i onClick={this.incrementVotes.bind(this)} className="fa fa-caret-up" />
+                </div>
+                <div>
+                    {this.state.votes}
+                </div>
+                <div className="vote">
+                    <i onClick={this.decrementVotes.bind(this)} className="fa fa-caret-down" />
+                </div>
+            </>
+        );
+    }
+
+    renderButtons() {
+        if (this.props.userId !== this.context.state.user.id) {
+            return null;
+        }
+
+        return (
+            <div className="button-group">
+                <Button color="secondary" text small onClick={this.editPost.bind(this)}>
+                    <i className="fa fa-pencil" /> {t`Edit`}
+                </Button>
+                <Button color="secondary" text small onClick={this.openDeleteDialog.bind(this)}>
+                    <i className="fa fa-trash" /> {t`Delete`}
+                </Button>
+            </div>
+        );
+    }
+
+    render() {
+        return (
+            <div className={this.getClasses()}>
```

Comment:
Do we think that we can reduce the size of this render into smaller parts? or is it easier to read this way?

Reply:
Fixed